src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/ItemManager.CompletionListUpdater.cs

```diff
@@ -0,0 +1,805 @@
+                        : ImmutableArray<Span>.Empty;
+
+                    return new CompletionItemWithHighlight(matchResult.EditorCompletionItem, highlightedSpans);
+                });
+
+                static ImmutableArray<Span> GetHighlightedSpans(
+                    MatchResult<VSCompletionItem> matchResult,
+                    CompletionHelper completionHelper,
+                    string filterText)
+                {
+                    if (matchResult.RoslynCompletionItem.HasDifferentFilterText)
+                    {
+                        // The PatternMatch in MatchResult is calculated based on Roslyn item's FilterText, 
+                        // which can be used to calculate highlighted span for VSCompletion item's DisplayText w/o doing the matching again.
+                        // However, if the Roslyn item's FilterText is different from its DisplayText,
+                        // we need to do the match against the display text of the VS item directly to get the highlighted spans.
+                        return completionHelper.GetHighlightedSpans(
+                            matchResult.EditorCompletionItem.DisplayText, filterText, CultureInfo.CurrentCulture).SelectAsArray(s => s.ToSpan());
+                    }
+
+                    var patternMatch = matchResult.PatternMatch;
+                    if (patternMatch.HasValue)
+                    {
+                        // Since VS item's display text is created as Prefix + DisplayText + Suffix, 
+                        // we can calculate the highlighted span by adding an offset that is the length of the Prefix.
+                        return patternMatch.Value.MatchedSpans.SelectAsArray(s_highlightSpanGetter, matchResult.RoslynCompletionItem);
+                    }
+
+                    // If there's no match for Roslyn item's filter text which is identical to its display text,
+                    // then we can safely assume there'd be no matching to VS item's display text.
+                    return ImmutableArray<Span>.Empty;
+                }
+            }
+
+            private FilteredCompletionModel? HandleAllItemsFilteredOut()
+            {
+                if (UpdateTriggerReason == CompletionTriggerReason.Insertion)
+                {
+                    // If the user was just typing, and the list went to empty *and* this is a 
+                    // language that wants to dismiss on empty, then just return a null model
+                    // to stop the completion session.
+                    if (_completionRules.DismissIfEmpty)
+                    {
+                        return null;
+                    }
+                }
+
+                // If the user has turned on some filtering states, and we filtered down to
+                // nothing, then we do want the UI to show that to them.  That way the user
+                // can turn off filters they don't want and get the right set of items.
```

Comment:
this comment doesnt' seem to match the logic in the method.  we seem to unilaterally bail out above, even if filters are set.